pkg/kubectl/cmd/wait/wait_test.go

```diff
@@ -298,6 +298,64 @@ func TestWaitForDeletion(t *testing.T) {
 				}
 			},
 		},
+		{
+			name: "handles watch delete multiple",
+			infos: []*resource.Info{
+				{
+					Mapping: &meta.RESTMapping{
+						Resource: schema.GroupVersionResource{Group: "group", Version: "version", Resource: "theresource-1"},
+					},
+					Name:      "name-foo-1",
+					Namespace: "ns-foo",
+				},
+				{
+					Mapping: &meta.RESTMapping{
+						Resource: schema.GroupVersionResource{Group: "group", Version: "version", Resource: "theresource-2"},
+					},
+					Name:      "name-foo-2",
+					Namespace: "ns-foo",
+				},
+			},
+			fakeClient: func() *dynamicfakeclient.FakeDynamicClient {
+				fakeClient := dynamicfakeclient.NewSimpleDynamicClient(scheme)
+				fakeClient.PrependReactor("get", "theresource-1", func(action clienttesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, newUnstructured("group/version", "TheKind", "ns-foo", "name-foo-1"), nil
+				})
+				fakeClient.PrependReactor("get", "theresource-2", func(action clienttesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, newUnstructured("group/version", "TheKind", "ns-foo", "name-foo-2"), nil
+				})
+				fakeClient.PrependWatchReactor("theresource-1", func(action clienttesting.Action) (handled bool, ret watch.Interface, err error) {
+					fakeWatch := watch.NewRaceFreeFake()
+					fakeWatch.Action(watch.Deleted, newUnstructured("group/version", "TheKind", "ns-foo", "name-foo-1"))
+					return true, fakeWatch, nil
+				})
+				fakeClient.PrependWatchReactor("theresource-2", func(action clienttesting.Action) (handled bool, ret watch.Interface, err error) {
+					fakeWatch := watch.NewRaceFreeFake()
+					fakeWatch.Action(watch.Deleted, newUnstructured("group/version", "TheKind", "ns-foo", "name-foo-2"))
+					return true, fakeWatch, nil
+				})
+				return fakeClient
+			},
+			timeout: 10 * time.Second,
+
+			validateActions: func(t *testing.T, actions []clienttesting.Action) {
+				if len(actions) != 4 {
+					t.Fatal(spew.Sdump(actions))
+				}
+				if !actions[0].Matches("get", "theresource-1") || actions[0].(clienttesting.GetAction).GetName() != "name-foo-1" {
+					t.Error(spew.Sdump(actions))
+				}
+				if !actions[1].Matches("watch", "theresource-1") {
+					t.Error(spew.Sdump(actions))
+				}
+				if !actions[2].Matches("get", "theresource-2") || actions[2].(clienttesting.GetAction).GetName() != "name-foo-2" {
+					t.Error(spew.Sdump(actions))
+				}
+				if !actions[3].Matches("watch", "theresource-2") {
+					t.Error(spew.Sdump(actions))
+				}
+			},
+		},
```

Comment:
This test doesn't prove `--all` is wired properly, it'll pass with and without your change.

Reply:
This test proves that the wait command is able to await multiple resources, which is what the use-case that `--all` expands to. Neither of these tests are checking for flag-parsing as this is not made in the wait package but is a part of `genericclioptions`. Flag parsing should thus not be tested here as it is covered under `k8s.io/cli-runtime/pkg/genericclioptions`. This test should be here, as there was no previous test ensuring that `wait` was able to handle multiple resources.

Reply:
I'm not saying this test is wrong, I'm just saying it doesn't ensure `--all` is wired well. I'm also aware we're missing flag wiring PRs, so either add one here or add one to test-cmd suite. We need to ensure something that will ensure that `--all` works as expected.

Reply:
As discussed on Slack, we will add test for `kubectl wait` in `test-cmd` to test for proper flag parsing. I will amend the commit with these changes once I have them :)